### ISSUE TITLE
Assertion failed 'this->Prev()->KindIs(BBJ_CALLFINALLY)'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/runtime#97231]
x86 JitStress is broken.

https://dev.azure.com/dnceng-public/public/_build/results?buildId=532905&view=ms.vss-test-web.build-test-results-tab
https://dev.azure.com/dnceng-public/public/_build/results?buildId=532976&view=ms.vss-test-web.build-test-results-tab

```
12:35:01.513 Running test: JIT/Directed/coverage/oldtests/Desktop/callipinvoke_d/callipinvoke_d.cmd

Assert failure(PID 1748 [0x000006d4], Thread: 3076 [0x0c04]): Assertion failed 'this->Prev()->KindIs(BBJ_CALLFINALLY)' in 'System.Threading.ThreadPoolWorkQueue+WorkStealingQueue:LocalPopCore():System.Object:this' during 'Find loops' (IL size 223; hash 0xd0f57bc2; Instrumented Tier1)

    File: D:\a\_work\1\s\src\coreclr\jit\block.cpp Line: 1711
    Image: C:\h\w\BB050A3E\p\corerun.exe

App Exit Code: -1073740286
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
```

@dotnet/jit-contrib

**Comment by jakobbotsch** [comment]
Hmm, odd.. that's what [this](https://github.com/dotnet/runtime/pull/97149#discussion_r1457965189) fix was about. I'll take a look, I must have missed some case.

**Comment by jakobbotsch** [comment]
Ah, the issue was introduced by ##96995 and fixed by #97149. The failing jitstress run happened between the two PRs.
